src/query-builder/builder/DeleteQueryBuilder.ts

```diff
@@ -0,0 +1,90 @@
+import {DeleteResult} from "../result/DeleteResult";
+import {MysqlDriver} from "../../driver/mysql/MysqlDriver";
+import {BroadcasterResult} from "../../subscriber/BroadcasterResult";
+import { EntitySchema} from "../../index";
+import {AuroraDataApiDriver} from "../../driver/aurora-data-api/AuroraDataApiDriver";
+import {BetterSqlite3Driver} from "../../driver/better-sqlite3/BetterSqlite3Driver";
+import { ModificationQueryBuilder } from "./ModificationQueryBuilder";
+
+/**
+ * Allows to build complex sql queries in a fashion way and execute those queries.
+ */
+export class DeleteQueryBuilder<Entity> extends ModificationQueryBuilder<Entity, DeleteResult> {
+
+    // -------------------------------------------------------------------------
+    // Public Methods
+    // -------------------------------------------------------------------------
+
+    /**
+     * Specifies FROM which entity's table select/update/delete will be executed.
+     * Also sets a main string alias of the selection data.
+     */
+    from<T>(entityTarget: EntityTarget<T>, aliasName?: string): DeleteQueryBuilder<T> {
+        entityTarget = entityTarget instanceof EntitySchema ? entityTarget.options.name : entityTarget;
+        const mainAlias = this.createFromAlias(entityTarget, aliasName);
+        this.expressionMap.setMainAlias(mainAlias);
+        return (this as any) as DeleteQueryBuilder<T>;
+    }
+
+    // -------------------------------------------------------------------------
+    // Protected Implemented Methods
+    // -------------------------------------------------------------------------
+
+    /**
+     * Creates DELETE expression used to perform query.
+     */
+    protected createModificationExpression() {
+        const tableName = this.getTableName(this.getMainTableName());
+        const whereExpression = this.createWhereExpression();
+        const returningExpression = this.createReturningExpression();
+
+        if (returningExpression && (this.connection.driver instanceof PostgresDriver || this.connection.driver instanceof CockroachDriver)) {
+            return `DELETE FROM ${tableName}${whereExpression} RETURNING ${returningExpression}`;
+        } else if (returningExpression !== "" && this.connection.driver instanceof SqlServerDriver) {
+            return `DELETE FROM ${tableName} OUTPUT ${returningExpression}${whereExpression}`;
+        } else {
+            return `DELETE FROM ${tableName}${whereExpression}`;
+        }
+    }
+
+    /**
+     * Executes sql generated by query builder and returns raw database results.
+     */
+    protected async executeInsideTransaction(queryRunner: QueryRunner): Promise<DeleteResult> {
+        const [sql, parameters] = this.getQueryAndParameters();
+        const deleteResult = new DeleteResult();
+        const result = await queryRunner.query(sql, parameters);
+
+        if (this.connection.driver instanceof MysqlDriver || this.connection.driver instanceof AuroraDataApiDriver) {
```

Comment:
Can this be moved to drivers or `QueryRunner`? Each driver should know its expected return formats.

Reply:
Also with a generic solution the Insert/Update/Delete builders would become even more similar in this function.

Reply:
But more generally something should really be done about the `instanceof Driver` spam...